### PR TITLE
CloudWatch: Add AWS/S3 replication metrics (#74416)

### LINKED ISSUE #74416: Plugins: Unable to get CloudWatch AWS/S3 replication metrics
### What happened?

I'm not able to get  "OperationsPendingReplication", "BytesPendingReplication", "ReplicationLatency", "OperationsFailedReplication"  in AWS/S3 namespace metrics list.

### What did you expect to happen?

I want to query  AWS/S3 replication metrics

### Did this work before?

No

### How do we reproduce it?

1. In Explorer, select the cloudwatch plugin as datasource
2. Select the AWS/S3 Namespace
3. Replication related metrics are not listed

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: v9.3.8
OS:
Browser:

### Grafana platform?

Kubernetes

### Datasource(s)?

Cloudwatch

### PR DESCRIPTION
**Pull request by jordanefillatre**
* Add S3 replications metrics to appropriate Namespace in NamespaceMetricsMap

**What is this feature?**

Allow to use AWS S3 replication metrics using Cloudwatch datasource 

**Why do we need this feature?**

Query "OperationsPendingReplication", "BytesPendingReplication", "ReplicationLatency", "OperationsFailedReplication" metrics

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #74416

**Special notes for your reviewer:**

Please check that:
- [ x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by iwysiu — COMMENTED** [review]
Hi @jordanefillatre ! Thanks for the contribution. Would you mind signing the CLA and reordering the new metrics to be in alphabetical order with the old ones?

Adding a link to the documentation: https://docs.aws.amazon.com/AmazonS3/latest/userguide/metrics-dimensions.html
Have you tried it and confirmed that it works? I noticed that the documentation didn't explicitly list the namespace

**Comment by jordanefillatre** [comment]
Hi @iwysiu ,
You're right, the documentation isn't very clear about namespace. 
I have confirm it by querying api using aws cli:

```
            "MetricName": "OperationsFailedReplication",
            "Dimensions": [
                {
                    "Name": "SourceBucket",
                    "Value": "xxx"
                },
                {
                    "Name": "DestinationBucket",
                    "Value": "xxx"
                },
                {
                    "Name": "RuleId",
                    "Value": "crr-full-standard"
                }
            ]
        },
```
It also show that I have omitted  to add required dimensions. 

Last I've take more time to manually check manually the fix by running a dev environment. I can confirm that it works after last commit update:

Capture d’écran du 2023-09-06 09-56-13

File: pkg/tsdb/cloudwatch/constants/metrics.go [hunk 1]
```diff
@@ -378,7 +378,7 @@ var NamespaceMetricsMap = map[string][]string{
 	"AWS/Robomaker":            {"RealTimeFactor", "vCPU", "Memory", "SimulationUnit"},
 	"AWS/Route53":              {"ChildHealthCheckHealthyCount", "ConnectionTime", "DNSQueries", "HealthCheckPercentageHealthy", "HealthCheckStatus", "SSLHandshakeTime", "TimeToFirstByte"},
 	"AWS/Route53Resolver":      {"InboundQueryVolume", "OutboundQueryVolume", "OutboundQueryAggregatedVolume"},
-	"AWS/S3":                   {"4xxErrors", "5xxErrors", "AllRequests", "BucketSizeBytes", "BytesDownloaded", "BytesUploaded", "DeleteRequests", "FirstByteLatency", "GetRequests", "HeadRequests", "ListRequests", "NumberOfObjects", "PostRequests", "PutRequests", "SelectRequests", "SelectReturnedBytes", "SelectScannedBytes", "TotalRequestLatency"},
+	"AWS/S3":                   {"4xxErrors", "5xxErrors", "AllRequests", "BucketSizeBytes", "BytesDownloaded", "BytesPendingReplication", "BytesUploaded", "DeleteRequests", "FirstByteLatency", "GetRequests", "HeadRequests", "ListRequests", "NumberOfObjects", "PostRequests", "PutRequests", "OperationsFailedReplication", "OperationsPendingReplication", "ReplicationLatency", "SelectRequests", "SelectReturnedBytes", "SelectScannedBytes", "TotalRequestLatency"},
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/constants/metrics.go:381]: just fixing the ordering:
```suggestion
	"AWS/S3":                   {"4xxErrors", "5xxErrors", "AllRequests", "BucketSizeBytes", "BytesDownloaded", "BytesPendingReplication", "BytesUploaded", "DeleteRequests", "FirstByteLatency", "GetRequests", "HeadRequests", "ListRequests", "NumberOfObjects", "OperationsFailedReplication", "OperationsPendingReplication", "PostRequests", "PutRequests", "ReplicationLatency", "SelectRequests", "SelectReturnedBytes", "SelectScannedBytes", "TotalRequestLatency"},
```

**Review by iwysiu — COMMENTED** [review]
Looks good! Thanks for fixing the Dimensions and checking the namespace. Just one little alphabetization fix.

**Comment by jordanefillatre** [comment]
Oh, so sorry

**Review by iwysiu — APPROVED** [review]
No problem, I'll finish up the process stuff and then merge it. Thanks again for contributing!

**Pull request merged**